frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -125,45 +216,127 @@ const RoutingLabel: React.FC<RoutingLabelProps> = ({ labels }) => {
   return <div className="co-m-label co-m-label--expand">{list}</div>;
 };
 
+const deleteReceiver = (receiverName: string) => {
+  // remove any routes which use receiverToDelete
+  _.update(config, 'route.routes', (routes) => {
+    _.remove(routes, (route: AlertManagerRoute) => route.receiver === receiverName);
+    return routes;
+  });
+  // delete receiver
+  _.update(config, 'receivers', (receivers) => {
+    _.remove(receivers, (receiver: AlertManagerReceiver) => receiver.name === receiverName);
+    return receivers;
+  });
+  return patchAlertManagerConfig(secret, config).then(() => {
+    history.push('/monitoring/alertmanagerconfig');
+  });
+};
+
+const editReceiverMenuItem = (receiverName: string, canUseEditForm: boolean) => {
+  const menuItem = {
+    label: `Edit ${canUseEditForm ? 'Receiver' : 'YAML'}`,
+  };
+  _.set(menuItem, 'callback', () => {
+    const targetUrl = canUseEditForm
+      ? `/monitoring/alertmanagerconfig/receivers/${receiverName}/edit`
+      : `/monitoring/alertmanageryaml`;
+    return history.push(targetUrl);
+  });
+  return menuItem;
+};
+
+const deleteReceiverMenuItem = (
+  receiverName: string,
+  isDefaultReceiver: boolean,
+  canDelete: boolean,
+) => ({
+  label: 'Delete Receiver',
+  isDisabled: isDefaultReceiver || !canDelete,
+  tooltip:
+    isDefaultReceiver || !canDelete
+      ? 'Cannot delete the default receiver, or a receiver which has a sub-route'
+      : '',
+  callback: () =>
+    confirmModal({
+      title: 'Delete Receiver',
+      message: `Are you sure you want to delete receiver '${receiverName}' ?`,
+      btnText: 'Delete Receiver',
+      executeFn: () => deleteReceiver(receiverName),
+    }),
+});
+const receiverMenuActions = (
+  receiverName: string,
+  isDefaultReceiver: boolean,
+  canUseEditForm: boolean,
+  canDelete: boolean,
+) => [
+  editReceiverMenuItem(receiverName, canUseEditForm),
+  deleteReceiverMenuItem(receiverName, isDefaultReceiver, canDelete),
+];
+const ReceiverKebab = ({ receiverName, isDefaultReceiver, canUseEditForm, canDelete }) => (
+  <Kebab
+    options={receiverMenuActions(receiverName, isDefaultReceiver, canUseEditForm, canDelete)}
+  />
+);
+
 const ReceiverTableRow: React.FC<ReceiverTableRowProps> = ({
   obj: receiver,
   index,
   key,
   style,
-  customData: routingLabels,
+  customData,
 }) => {
+  const { routingLabelsByReceivers, defaultReceiverName } = customData;
   // filter to routing labels belonging to current Receiver
-  const receiverRoutingLabels = _.filter(routingLabels, { receiver: receiver.name });
+  const receiverRoutingLabels = _.filter(routingLabelsByReceivers, { receiver: receiver.name });
+  const integrationTypes = getIntegrationTypes(receiver);
+  const canUseEditForm = isValidForEditForm(receiver, receiverRoutingLabels, integrationTypes);
+  const canDelete = isValidForDelete(receiver, receiverRoutingLabels);
```

Comment:
I feel that having separate `isValidForEditForm` and `isValidForDelete` functions that are only called from here actually makes it harder to follow. Maybe just.
```
// Receiver form can only handle simple configurations. Can edit via form if receiver
// has a simple route and receiver
const canUseEditForm = hasSimpleRoute(receiver, receiverRoutingLabels) && hasSimpleReceiver(receiver);

// Receivers can be deleted if it has a simple route
const canDelete = hasSimpleRoute(receiver, receiverRoutingLabels);
```

Reply:
ok, fixed